src/fides/api/util/tcf/experience_meta.py

```diff
@@ -46,18 +63,74 @@ def _build_tcf_version_hash_model(
     get the maximum possible number of attributes added to our hash for the current
     system configuration.
     """
+
+    def build_vendor_by_purpose_strings(
+        vendor_list: Union[
+            List[TCFVendorConsentRecord], List[TCFVendorLegitimateInterestsRecord]
+        ]
+    ) -> List[str]:
+        """Hash helper for building vendor x purposes.  Converts a list of vendor records
+        into a list of strings each consisting of a vendor id, a separator (-), and comma
+        separated purposes (if applicable) with the given legitimate interest.
+
+        Example:
+        [<universal_vendor_id>-<comma separated purposes>]
+        ["gacp.100", "gvl.8-1,2,3", "gvl.56-3,4,5]
+        """
+        vendor_by_purpose_lists: List[str] = []
+        for vendor_record in vendor_list:
+            vendor_id: str = vendor_record.id
+            purposes_key: str = (
+                "purpose_consents"
+                if isinstance(vendor_record, TCFVendorConsentRecord)
+                else "purpose_legitimate_interests"
+            )
+            purpose_ids = sorted(
+                purpose_record.id
+                for purpose_record in getattr(vendor_record, purposes_key)
+            )
+            concatenated_purposes_string: str = ",".join(
+                str(purpose_id) for purpose_id in purpose_ids
+            )
+            vendor_by_purpose_lists.append(
+                vendor_id + "-" + concatenated_purposes_string
+                if concatenated_purposes_string
+                else vendor_id
+            )
+        return vendor_by_purpose_lists
+
+    vendor_consents: List[str] = build_vendor_by_purpose_strings(
+        tcf_contents.tcf_vendor_consents + tcf_contents.tcf_system_consents
+    )
+    vendor_li: List[str] = build_vendor_by_purpose_strings(
+        tcf_contents.tcf_vendor_legitimate_interests
+        + tcf_contents.tcf_system_legitimate_interests
+    )
+
+    # Building the TC model to pull off the GVL Policy version and get vendors_disclosed
+    # which is a component of the TC string.
     model: TCModel = convert_tcf_contents_to_tc_model(
         tcf_contents, UserConsentPreference.opt_in
     )
-    return TCFVersionHash(**model.dict())
 
+    return TCFVersionHash(
+        policy_version=model.policy_version,
+        vendor_purpose_consents=vendor_consents,
+        vendor_purpose_legitimate_interests=vendor_li,
+        gvl_vendors_disclosed=model.vendors_disclosed,
```

Comment:
may be worth commenting here that this is specifically GVL vendors that show up in the "current" GVL reference list. i know that can be found by tracing back and digging into the docstrings/code that is responsible for populating  `model.vendors_disclosed` (i did it!), that's a key nuance here that could easily get overlooked or forgotten about.

```suggestion
        gvl_vendors_disclosed=model.vendors_disclosed, # `vendors_disclosed` field only has GVL vendors present in "current" GVL vendor list
```

Reply:
I can add this 👍